### PR TITLE
[Core] added default factory for avatar image

### LINKED ISSUE #11163: AdminUserExampleFactory fails after AvatarImage overriden
**Sylius version affected**: 1.7 / any

**Description**  
Once `AvatarImage` overriden, `AdminUserExampleFactory` fails:

```bash
$ bin/console sylius:fixtures:load 
...
Running fixture "admin_user"...  

In AbstractMySQLDriver.php line 103:
                                                                               
  An exception occurred while executing 'INSERT INTO sylius_avatar_image (typ  
  e, path) VALUES (?, ?)' with params [null, "82\/5a\/96c5f765e53fd4c4ebb9a34  
  f6733.jpeg"]:                                                                
                                                                               
  SQLSTATE[HY000]: General error: 1364 Field 'owner_id' doesn't have a defaul  
  t value     
```

As far as `createAvatar` method uses direct entity creation I guess rather than factory:

```php
$avatarImage = new AvatarImage();
```

**Possible Solution**  
Use AvatarImage factory

### PR DESCRIPTION
**Pull request by foxou33**
| Q               | A
| --------------- | -----
| Branch?         | 1.7
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | fixes #11163 
| License         | MIT

### CODE REVIEW

File: src/Sylius/Component/Core/Factory/AvatarImageFactory.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Factory;
+
+use Sylius\Component\Core\Model\ImageInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+
+class AvatarImageFactory implements FactoryInterface
```
Comment (lchrusciel) [inline src/Sylius/Component/Core/Factory/AvatarImageFactory.php:19]: This class is not needed

Comment (foxou33) [reply src/Sylius/Component/Core/Factory/AvatarImageFactory.php]: I thought so too :)

**Comment by lchrusciel** [comment]
The build should be green after merging #11780

File: src/Sylius/Bundle/CoreBundle/Fixture/Factory/AdminUserExampleFactory.php [hunk 2]
```diff
@@ -44,11 +47,13 @@ class AdminUserExampleFactory extends AbstractExampleFactory implements ExampleF
 
     public function __construct(
         FactoryInterface $userFactory,
+        FactoryInterface $avatarImageFactory,
```
Comment (lchrusciel) [inline src/Sylius/Bundle/CoreBundle/Fixture/Factory/AdminUserExampleFactory.php:50]: This service should be injected the same way as `FileLocatorInterface` & `ImageUploaderInterface`. Also placed on the last line, to support BC.

File: src/Sylius/Bundle/CoreBundle/Fixture/Factory/AdminUserExampleFactory.php [hunk 3]
```diff
@@ -124,7 +129,8 @@ private function createAvatar(AdminUserInterface $adminUser, array $options): vo
         $imagePath = $this->fileLocator->locate($options['avatar']);
         $uploadedImage = new UploadedFile($imagePath, basename($imagePath));
 
-        $avatarImage = new AvatarImage();
+        /** @var AvatarImage $avatarImage */
+        $avatarImage = $this->avatarImageFactory->createNew();
```
Comment (lchrusciel) [inline src/Sylius/Bundle/CoreBundle/Fixture/Factory/AdminUserExampleFactory.php:133]: Once `$this->avatarImageFactory` will be nullable, we need to fallback to `new AvatarImage()` if the factory will not be inserted

File: src/Sylius/Bundle/CoreBundle/Fixture/Factory/AdminUserExampleFactory.php (hunk 2 above)
Comment (foxou33) [reply src/Sylius/Bundle/CoreBundle/Fixture/Factory/AdminUserExampleFactory.php:50]: Ah ok I'll do that ASAP

**Comment by foxou33** [comment]
I think I messed up trying to merge #11780. I'm not sure how to revert.

**Comment by lchrusciel** [comment]
I've rebased your PR and fixed commits. All of them are still yours, of course :) Let's wait for green build and it is good to go

**Review by lchrusciel — APPROVED** [review]

**Comment by igormukhingmailcom** [comment]
Not stale. Very useful and looks ready to merge, isn't it?

**Comment by igormukhingmailcom** [comment]
Thanks, @foxou33  and @lchrusciel :)

**Comment by lchrusciel** [comment]
Thank you, @foxou33! :tada:

**Pull request merged**